assets/js/lcp-beacon.js

```diff
@@ -1,152 +1,217 @@
+		}
+		if (!full_bg_prop || full_bg_prop === "") {
+			return null;
+		}
+
+		const matches = [...full_bg_prop.matchAll(css_bg_url_rgx)];
+		element_info.bg_set = matches.map(m => m[1] ? {src: m[1].trim() + (m[2] ? " " + m[2].trim() : "")} : {});
+		// Check if bg_set array is populated with empty objects
+		if (element_info.bg_set.every(item => item.src === "")) {
+			// If bg_set array is populated with empty objects, populate it with the URLs from the matches array
+			element_info.bg_set = matches.map(m => m[1] ? {src: m[1].trim()} : {});
+		}
+
+		if (element_info.bg_set.length > 0) {
+			element_info.src = element_info.bg_set[0].src;
+			if (element_info.type === "bg-img-set") {
+				element_info.src = element_info.bg_set;
+			}
+		}
+	}
+
+	return element_info;
 }
 
 let performance_images = [];
 
-async function main() {
-    // Filter the array based on the condition imageURL is not null
-    const filteredArray = LCPCandidates(1)
-    if (filteredArray.length !== 0) {
-        console.log("Estimated LCP element:", filteredArray);
-        performance_images = filteredArray.map((item) => ({
-            src: item.imageURL,
-            label: "lcp",
-        }));
-    } else {
-        console.log("No LCP candidate found.");
-    }
-
-    var above_the_fold_images = document.querySelectorAll("img");
-
-    for (var i = 0; i < above_the_fold_images.length; i++) {
-        var image = above_the_fold_images[i];
-        var rect = image.getBoundingClientRect();
-        var intersecting = isIntersecting(rect);
-        if (intersecting) {
-            var parent = image.parentNode;
-            while (parent !== document) {
-                var displayStyle = window.getComputedStyle(parent).display;
-                var visibilityStyle =
-                    window.getComputedStyle(parent).visibility;
-                if (displayStyle === "none" || visibilityStyle === "hidden") {
-                    break;
-                }
-                parent = parent.parentNode;
-            }
-            const isDuplicate = performance_images.some(
-                (item) => item.src === image.src
-            );
-
-            // If it's not a duplicate, push the new element
-            if (!isDuplicate && parent === document) {
-                performance_images.push({
-                    src: image.src,
-                    label: "above-the-fold",
-                });
-            }
-        }
-    }
-    console.log(performance_images);
-    var performance_images_json = JSON.stringify(performance_images);
-    window.performance_images_json = performance_images_json;
-    console.log(performance_images_json);
+function main() {
+	// Use LCPCandidates function to get all the elements in the viewport
+	const above_the_fold_images = LCPCandidates(Infinity);
+
+	if (above_the_fold_images.length !== 0) {
+		// Get the first element in the viewport and use as LCP
+		performance_images = [{
+			...above_the_fold_images[0].elementInfo,
```

Comment:
I don't think this will work. 
We can't take the first element of `above_the_fold_images` as it isn't always an element with an image. So we now get many LCPs as `null`. 

If I take a concrete example, on [this page](https://e2e.rocketlabsqa.ovh/lcp_bg_inline_template/) we get null because `above_the_fold_images` contains the following: 
```
atf image element:
0: {element: div.wrap, elementInfo: null}
1: {element: div#primary.content-area, elementInfo: null}
2: {element: main#main.site-main, elementInfo: null}
3: {element: div, elementInfo: Object}
4: {element: div, elementInfo: Object}
5: {element: p, elementInfo: null}
```

As you sort the results by their areas [here](https://github.com/wp-media/wp-rocket/pull/6579/files#diff-9f86fce58cce5e3f8244c7d9d080582dd4b3c6483896fba8a4ef73eabc5e5381R56) that's normal that `div#wrap` is first, as it wrap all content of the page. But that's not the div containing images.

Reply:
And while thinking about it, 

We should modify your whole condition by:
``` javascript
	const firstElementWithInfo = above_the_fold_images.find(item => item.elementInfo !== null);

	if (firstElementWithInfo) {
		performance_images = [{
			...firstElementWithInfo.elementInfo,
			label: "lcp",
		}];
	} else {
		console.log("No LCP candidate found.");
	}
```